### PR TITLE
Tests for room tags and push rules after upgrading a remote room

### PR DESCRIPTION
**Pull request by anoadragon453**
Synapse PR: https://github.com/matrix-org/synapse/pull/6155

Requires: https://github.com/matrix-org/sytest/pull/726

New versions of the existing `/upgrade copies push rules to the new room` and `/upgrade on a remote room copies user tags to the new room` tests, but testing with a user who is on a different server than the one that upgraded the room.

### CODE REVIEW

**Comment by anoadragon453** [comment]
This is ready to review once #726 lands.

File: tests/30rooms/60version_upgrade.pl [hunk 1]
```diff
@@ -455,58 +455,74 @@ sub upgrade_room_synced {
...
-      })->then( sub {
-         ( $new_room_id, ) = @_;
+         matrix_invite_user_to_room_synced(
+            $creator, $joiner, $room_id,
+         )->then( sub {
+            matrix_join_room_synced(
+               $joiner, $room_id, ( server_name => $creator->server_name, ),
+            );
+         })->then( sub {
+            matrix_sync( $joiner );
```
Comment (erikjohnston) [inline tests/30rooms/60version_upgrade.pl]: Why the sync?

File: tests/30rooms/60version_upgrade.pl [hunk 2]
```diff
@@ -455,58 +455,74 @@ sub upgrade_room_synced {
...
+         })->then( sub {
+            matrix_add_push_rule(
+               $joiner, "global", "room", $room_id,
+               { actions => [ "notify" ] },
+            );
+         })->then(sub {
+            matrix_sync( $joiner );
```
Comment (erikjohnston) [inline tests/30rooms/60version_upgrade.pl]: Seems pointless?

File: tests/42tags.pl [hunk 3]
```diff
@@ -408,40 +408,98 @@ sub check_account_data {
...
+         )->then( sub {
+            matrix_join_room_synced(
+               $joiner, $room_id, ( server_name => $creator->server_name, ),
+            );
+         })->then( sub {
+            matrix_add_tag(
+               $joiner, $room_id, "test_tag", {}
+            );
+         })->then(sub {
+            matrix_sync( $joiner );
```
Comment (erikjohnston) [inline tests/42tags.pl]: etc...

File: tests/42tags.pl [hunk 4]
```diff
@@ -408,40 +408,98 @@ sub check_account_data {
...
+
+            keys %{$tags} == 1 or die "Expected one tag in the upgraded room";
+            defined $tags->{test_tag} or die "Unexpected tag";
+
+            Future->done(1);
+         });
+      }
+}
+
+test "/upgrade on a remote room copies user tags to the new room",
```
Comment (erikjohnston) [inline tests/42tags.pl]: Isn't this the same test as above?

**Pull request merged**